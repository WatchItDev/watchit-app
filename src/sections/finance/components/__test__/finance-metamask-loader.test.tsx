import { describe, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";
import FinanceMetamaskLoader from "../finance-metamask-loader";

describe("<COMPONENTS> FinanceMetamaskLoader", () => {
  it("to match snapshot", () => {
    const { container } = render(<FinanceMetamaskLoader />);
    expect(container).toMatchSnapshot();
  });

  it("should render the LoadingScreen component", () => {
    render(<FinanceMetamaskLoader />);
    const loadingScreen = screen.getByTestId("finance-metamask-loader");
    expect(loadingScreen).toBeInTheDocument();
  });

  it("should apply correct Box styles", () => {
    const { container } = render(<FinanceMetamaskLoader />);
    const boxElement = container.querySelector("div");
    expect(boxElement).toHaveStyle("margin-left: 32px");
    expect(boxElement).toHaveStyle("margin-right: 32px");
    expect(boxElement).toHaveStyle("margin-top: 64px");
    expect(boxElement).toHaveStyle("margin-bottom: 64px");
  });
});
