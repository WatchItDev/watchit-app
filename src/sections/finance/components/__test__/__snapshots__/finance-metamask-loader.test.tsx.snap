// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<COMPONENTS> FinanceMetamaskLoader > to match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1dm8t1i"
  >
    <div
      class="MuiBox-root css-tk1iyj"
      data-testid="finance-metamask-loader"
    >
      <span
        class="MuiLinearProgress-root MuiLinearProgress-colorInherit MuiLinearProgress-indeterminate css-yptbrg-MuiLinearProgress-root"
        role="progressbar"
      >
        <span
          class="MuiLinearProgress-bar MuiLinearProgress-barColorInherit MuiLinearProgress-bar1Indeterminate css-jlm0hp-MuiLinearProgress-bar1"
        />
        <span
          class="MuiLinearProgress-bar MuiLinearProgress-barColorInherit MuiLinearProgress-bar2Indeterminate css-1wsh2fw-MuiLinearProgress-bar2"
        />
      </span>
    </div>
  </div>
</div>
`;
